src/pages/index.js

```diff
@@ -0,0 +1,363 @@
+  if (evt.target.classList.contains("modal")) {
+    closeModal(cardSubmitModal);
+  }
+});
+
+previewModal.addEventListener("mousedown", (evt) => {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(previewModal);
+  }
+});
+
+function setUserData(data) {
+  profileTitle.textContent = data.name;
+  profileDescription.textContent = data.about;
+  profileAvatar.src = data.avatar;
+}
+
+function handleEditFormSubmit(evt) {
+  console.log("submit it");
+  evt.preventDefault();
+  editButtonEl.textContent = "Loading...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      setUserData(data);
+    })
+    .catch(console.error)
+
+    .finally(() => {
+      editButtonEl.textContent = "Save";
+    });
+
+  closeModal(editProfileModal);
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileTitle.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+
+  resetValidation(editFormElement, settings);
+  openModal(editProfileModal);
+});
+
+editProfileModalClosedBtn.addEventListener("click", () => {
+  closeModal(editProfileModal);
+});
+
+cardCloseBtn.addEventListener("click", () => {
+  closeModal(cardSubmitModal);
+});
+
+avatarCloseBtn.addEventListener("click", () => {
+  closeModal(avatarSubmitModal);
+});
+
+previewModalCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+const addCardButton = document.querySelector(".profile__add-btn");
+addCardButton.addEventListener("click", () => {
+  openModal(cardSubmitModal);
+});
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarSubmitModal);
+});
+// TODO - Don't forget form validation
+// TODO - Set up close button listener
+
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+cardFormElement.addEventListener("submit", handleAddCardFormSubmit);
+avatarFormElement.addEventListener("submit", handleAvatarSumbit);
+
+function getcardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  // TODO - If the card is liked, set the active class on the card
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Show whether card is liked or not
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    handleLike(cardElement, data); // Use API to update the card on the server
+    console.log("Card Liked");
+    cardLikeBtn.classList.toggle("card__like-btn_liked"); // Update teh UI
```

Comment:
` cardLikeBtn.classList.toggle("card__like-btn_liked"); // Update teh UI`

Please, delete it because it's done inside `handleLike` in a `.then` block

You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.